src/components/Editor/index.js

```diff
@@ -160,16 +164,35 @@ class Editor extends Component {
     } else {
       this.undoTimestamp = 0
     }
+
+    if (evt.keyCode === 27 && !this.state.tabGuarded) {// Esc Key
+      this.setState({tabGuarded:true})
+    }else if(!(evt.shiftKey || evt.keyCode === 27 || evt.keyCode === 9) && this.state.tabGuarded){
```

Comment:
In `keyup` the `shift` key (19) triggers separately when released. So here you'll need to add `evt.keyCode === 16` as well. Just discovered that :/

Also, can we do `this.state.tabGuarded && !evt.shiftKey && evt.keyCode !== 27 /* ... */` instead? i.e. negate the group there to remove it, and put the `tabGuarded` first? Should help readability a little.